File: MacOs/Shared/CategoryDefinitions.swift
import SwiftUI

// ─── Category definitions ─────────────────────────────────────────────────────

struct CategoryOption: Equatable {
    let label: String
    let emoji: String
    let description: String
    let starterNotes: String
    let accent: Color
}

let hobbyCategories: [CategoryOption] = [
    CategoryOption(label: "General", emoji: "📋", description: "Flexible trackers for anything recurring.",
        starterNotes: "What matters, how often it needs attention, and what a good update looks like.",
        accent: Color(hex: 0x4E5D6C)),
    CategoryOption(label: "Plants", emoji: "🌱", description: "Watering, pruning, fertilizer, sunlight, pests.",
        starterNotes: "Plant species, watering preference, light conditions, fertilizer schedule, and warning signs.",
        accent: Color(hex: 0x2D7A46)),
    CategoryOption(label: "Garden", emoji: "🌻", description: "Outdoor beds, lawn, compost, harvests, seasonal jobs.",
        starterNotes: "Plot/bed, current crops, watering days, fertilizer or compost cadence, and pest watch.",
        accent: Color(hex: 0x6E8B3D)),
    CategoryOption(label: "Fitness", emoji: "💪", description: "Workouts, mobility, recovery, steps, practice streaks.",
        starterNotes: "Routine, target frequency, current baseline, progress marker, and recovery notes.",
        accent: Color(hex: 0xB14E32)),
    CategoryOption(label: "Medicine", emoji: "💊", description: "Medications, dosages, refills, side effects.",
        starterNotes: "Drug name, dosage, schedule (morning/evening), refill date, and side effects to watch for.",
        accent: Color(hex: 0xD81B60)),
    CategoryOption(label: "Health", emoji: "❤️", description: "Doctor visits, sleep, hydration, vitals, symptoms.",
        starterNotes: "What to track (BP, weight, sleep, hydration), check-up cadence, and any doctor's notes.",
        accent: Color(hex: 0xC2185B)),
    CategoryOption(label: "Mindfulness", emoji: "🧘", description: "Meditation, journaling, breathing, gratitude.",
        starterNotes: "Practice length, technique, intentions, and how you felt before/after.",
        accent: Color(hex: 0x7E57C2)),
    CategoryOption(label: "Reading", emoji: "📚", description: "Books, chapters, notes, reading streaks.",
        starterNotes: "Current book, chapter target, themes to capture, and next reading checkpoint.",
        accent: Color(hex: 0x6F4A8E)),
    CategoryOption(label: "Learning", emoji: "🧠", description: "Courses, language, coding, certifications.",
        starterNotes: "Topic, resource, practice plan, review checkpoints, and questions to revisit.",
        accent: Color(hex: 0x4867A5)),
    CategoryOption(label: "Music", emoji: "🎵", description: "Instrument practice, drills, songs, recordings.",
        starterNotes: "Warmups, technique focus, piece/song, tempo target, and recording notes.",
        accent: Color(hex: 0x2E6F95)),
    CategoryOption(label: "Creative", emoji: "🎨", description: "Writing, sketching, photos, side projects.",
        starterNotes: "Project goal, next small action, reference ideas, and review cadence.",
        accent: Color(hex: 0xB05D6D)),
    CategoryOption(label: "Cooking", emoji: "🍳", description: "Recipes, meal prep, baking, fermentation, kitchen wins.",
        starterNotes: "Recipe or technique, ingredients, cooking time, taste notes, and what to change next.",
        accent: Color(hex: 0xE07A1F)),
    CategoryOption(label: "Coffee", emoji: "☕", description: "Beans, grind, recipes, brew ratios, tasting notes.",
        starterNotes: "Bean, roast date, grinder setting, dose, yield, time, and taste notes.",
        accent: Color(hex: 0x8A5A2B)),
    CategoryOption(label: "Pets", emoji: "🐾", description: "Feeding, walks, grooming, vet, training sessions.",
        starterNotes: "Pet name, feeding cadence, walk/play schedule, grooming routine, and vet milestones.",
        accent: Color(hex: 0x8D6E63)),
    CategoryOption(label: "Aquarium", emoji: "🐠", description: "Water tests, feeding, cleaning, livestock health.",
        starterNotes: "Tank size, parameters to test, feeding cadence, cleaning routine, and livestock observations.",
        accent: Color(hex: 0x207A8A)),
    CategoryOption(label: "Travel", emoji: "✈️", description: "Trips, packing, places visited, expense receipts.",
        starterNotes: "Destination, dates, packing list, must-see places, and cost log.",
        accent: Color(hex: 0x0097A7)),
    CategoryOption(label: "Gaming", emoji: "🎮", description: "Game progress, sessions, achievements, backlog.",
        starterNotes: "Title, current chapter/rank, target achievement, session length, and backlog notes.",
        accent: Color(hex: 0x5C6BC0)),
    CategoryOption(label: "Tech", emoji: "💻", description: "Devices, side projects, builds, deployments, learning logs.",
        starterNotes: "Project or device, current task, dependencies, blockers, and next milestone.",
        accent: Color(hex: 0x455A64)),
    CategoryOption(label: "Work", emoji: "💼", description: "Standups, deliverables, deep-work sessions, follow-ups.",
        starterNotes: "Goal, current task, blockers, time spent, and what's queued for tomorrow.",
        accent: Color(hex: 0x37474F)),
    CategoryOption(label: "Finance", emoji: "💰", description: "Bills, budgets, subscriptions, investments, savings goals.",
        starterNotes: "Account/bill name, due date or cadence, target amount, and balance trend.",
        accent: Color(hex: 0x388E3C)),
    CategoryOption(label: "Collecting", emoji: "🎒", description: "Sneakers, keyboards, watches, figures, catalog care.",
        starterNotes: "Item details, condition, purchase date, care routine, and rotation notes.",
        accent: Color(hex: 0x8B5E83)),
    CategoryOption(label: "Maintenance", emoji: "🔧", description: "Home, gear, bike, car, filters, renewals.",
        starterNotes: "Asset, recurring task, parts/tools, last service date, and next service trigger.",
        accent: Color(hex: 0x6B6D3F)),
]

func categoryFor(_ label: String) -> CategoryOption {
    let trimmed = label.trimmingCharacters(in: .whitespacesAndNewlines)
    if let match = hobbyCategories.first(where: { $0.label.caseInsensitiveCompare(trimmed) == .orderedSame }) {
        return match
    }
    let base = hobbyCategories[0]
    return CategoryOption(
        label: trimmed.isEmpty ? "General" : trimmed,
        emoji: "📋",
        description: "Custom tracker category.",
        starterNotes: "What to track, when to check in, and what a useful note should include.",
        accent: base.accent
    )
}

// ─── Reminder options ────────────────────────────────────────────────────────

struct ReminderOption: Identifiable, Equatable {
    var id: Int64 { hours }
    let label: String
    let hours: Int64
    let description: String
}

let reminderOptions: [ReminderOption] = [
    ReminderOption(label: "None", hours: 0, description: "No reminder"),
    ReminderOption(label: "1h", hours: 1, description: "Quick follow-up"),
    ReminderOption(label: "3h", hours: 3, description: "Later today"),
    ReminderOption(label: "6h", hours: 6, description: "Half day"),
    ReminderOption(label: "12h", hours: 12, description: "Tonight"),
    ReminderOption(label: "1d", hours: 24, description: "Tomorrow"),
    ReminderOption(label: "2d", hours: 48, description: "Two days"),
    ReminderOption(label: "3d", hours: 72, description: "Three days"),
    ReminderOption(label: "1w", hours: 168, description: "Weekly"),
    ReminderOption(label: "2w", hours: 336, description: "Bi-weekly"),
    ReminderOption(label: "30d", hours: 720, description: "Monthly"),
]

func reminderLabel(_ hours: Int64) -> String {
    reminderOptions.first(where: { $0.hours == hours })?.description ?? "\(hours)h"
}

// ─── Starter templates ───────────────────────────────────────────────────────

struct HobbyTemplate: Identifiable, Equatable {
    var id: String { title }
    let title: String
    let category: String
    let trackerName: String
    let notes: String
    let reminderHours: Int64
}

let hobbyTemplates: [HobbyTemplate] = [
    HobbyTemplate(title: "Plant care", category: "Plants", trackerName: "Monstera care",
        notes: "Track watering, leaf checks, pruning, fertilizer, and light changes.", reminderHours: 72),
    HobbyTemplate(title: "Daily meditation", category: "Mindfulness", trackerName: "Morning meditation",
        notes: "Capture practice length, technique, intention, and how you felt before/after.", reminderHours: 24),
    HobbyTemplate(title: "Medication", category: "Medicine", trackerName: "Daily meds",
        notes: "Track medications taken, missed doses, refill dates, and side effects.", reminderHours: 24),
    HobbyTemplate(title: "Refill calendar", category: "Medicine", trackerName: "Prescription refills",
        notes: "Each prescription's name, dosage, supply remaining, pharmacy, and next refill date.", reminderHours: 168),
    HobbyTemplate(title: "Sleep journal", category: "Health", trackerName: "Sleep tracker",
        notes: "Bedtime, wake time, sleep quality (1–5), dreams or wake-ups, energy in the morning.", reminderHours: 24),
    HobbyTemplate(title: "Coffee dial-in", category: "Coffee", trackerName: "Espresso recipe",
        notes: "Record bean, dose, yield, grind, shot time, and taste notes.", reminderHours: 0),
    HobbyTemplate(title: "Cooking journal", category: "Cooking", trackerName: "Weeknight dinners",
        notes: "Recipe, ingredients, cooking time, taste notes, and what to change next.", reminderHours: 0),
    HobbyTemplate(title: "Aquarium routine", category: "Aquarium", trackerName: "Tank maintenance",
        notes: "Log water parameters, feeding, filter checks, cleaning, and livestock health.", reminderHours: 168),
    HobbyTemplate(title: "Pet care", category: "Pets", trackerName: "Dog routine",
        notes: "Walks, feeding, training drills, grooming, and vet milestones.", reminderHours: 12),
    HobbyTemplate(title: "Practice streak", category: "Music", trackerName: "Guitar practice",
        notes: "Capture warmups, drills, song sections, tempo, and recordings.", reminderHours: 24),
    HobbyTemplate(title: "Gear rotation", category: "Collecting", trackerName: "Sneaker rotation",
        notes: "Track wears, cleaning, condition, weather, and storage notes.", reminderHours: 336),
    HobbyTemplate(title: "Study plan", category: "Learning", trackerName: "Android learning",
        notes: "Plan lessons, exercises, questions, review dates, and small projects.", reminderHours: 24),
    HobbyTemplate(title: "Reading list", category: "Reading", trackerName: "Currently reading",
        notes: "Current book, chapter target, themes to capture, and weekly reading goal.", reminderHours: 24),
    HobbyTemplate(title: "Side project", category: "Tech", trackerName: "Side project",
        notes: "Goal, current task, blockers, deploy/test cadence, next milestone.", reminderHours: 48),
    HobbyTemplate(title: "Workout split", category: "Fitness", trackerName: "Push/Pull/Legs",
        notes: "Split day, key lifts, weight progression, mobility, recovery score.", reminderHours: 24),
    HobbyTemplate(title: "Bills tracker", category: "Finance", trackerName: "Monthly bills",
        notes: "Bill name, due date, autopay, amount, and renewal notes.", reminderHours: 720),
    HobbyTemplate(title: "Gaming backlog", category: "Gaming", trackerName: "Backlog progress",
        notes: "Title, current chapter/rank, target achievement, and total play hours.", reminderHours: 168),
    HobbyTemplate(title: "Trip planner", category: "Travel", trackerName: "Next trip",
        notes: "Destination, dates, packing list, must-see places, and cost log.", reminderHours: 0),
    HobbyTemplate(title: "Garden bed", category: "Garden", trackerName: "Veggie bed",
        notes: "Bed, planted crops, watering days, fertilizer, and harvest notes.", reminderHours: 24),
    HobbyTemplate(title: "Daily journal", category: "Mindfulness", trackerName: "Daily journal",
        notes: "Three things, mood, gratitude, and one intention for tomorrow.", reminderHours: 24),
    HobbyTemplate(title: "Work standup", category: "Work", trackerName: "Daily standup",
        notes: "Done yesterday, doing today, blockers, and follow-ups owed.", reminderHours: 24),
]

// ─── Quick-log presets ───────────────────────────────────────────────────────

struct QuickLogPreset: Identifiable, Equatable {
    var id: String { label }
    let label: String
    let entry: String
}

func quickLogPresetsFor(_ category: String) -> [QuickLogPreset] {
    switch categoryFor(category).label {
    case "Plants":
        return [QuickLogPreset(label: "Watered", entry: "Watered and checked soil moisture."),
                QuickLogPreset(label: "Fertilized", entry: "Added fertilizer and noted plant response."),
                QuickLogPreset(label: "Leaf check", entry: "Checked leaves for pests, yellowing, and new growth.")]
    case "Fitness":
        return [QuickLogPreset(label: "Workout", entry: "Completed workout and logged effort level."),
                QuickLogPreset(label: "Mobility", entry: "Completed mobility or stretching session."),
                QuickLogPreset(label: "Recovery", entry: "Logged rest, soreness, sleep, or recovery notes.")]
    case "Reading":
        return [QuickLogPreset(label: "Read", entry: "Read a session and captured key takeaways."),
                QuickLogPreset(label: "Quote", entry: "Saved a memorable idea or quote."),
                QuickLogPreset(label: "Review", entry: "Reviewed notes and updated next reading target.")]
    case "Music":
        return [QuickLogPreset(label: "Practice", entry: "Completed practice session with focus notes."),
                QuickLogPreset(label: "Tempo", entry: "Worked on tempo, timing, or difficult section."),
                QuickLogPreset(label: "Recorded", entry: "Recorded a take and noted improvements.")]
    case "Coffee":
        return [QuickLogPreset(label: "Brewed", entry: "Logged brew recipe, extraction, and taste."),
                QuickLogPreset(label: "Adjusted", entry: "Adjusted grind, dose, yield, or brew time."),
                QuickLogPreset(label: "Tasted", entry: "Captured tasting notes and next change.")]
    case "Aquarium":
        return [QuickLogPreset(label: "Fed", entry: "Fed livestock and checked behavior."),
                QuickLogPreset(label: "Tested", entry: "Tested water parameters and logged results."),
                QuickLogPreset(label: "Cleaned", entry: "Completed cleaning, water change, or filter check.")]
    case "Collecting":
        return [QuickLogPreset(label: "Used", entry: "Logged use, wear, condition, and storage notes."),
                QuickLogPreset(label: "Cleaned", entry: "Cleaned item and checked condition."),
                QuickLogPreset(label: "Cataloged", entry: "Updated catalog details or purchase notes.")]
    case "Maintenance":
        return [QuickLogPreset(label: "Serviced", entry: "Completed maintenance task and noted parts/tools."),
                QuickLogPreset(label: "Inspected", entry: "Inspected condition and captured follow-up items."),
                QuickLogPreset(label: "Renewed", entry: "Updated renewal, warranty, or replacement details.")]
    case "Creative":
        return [QuickLogPreset(label: "Drafted", entry: "Moved the project forward with a focused work session."),
                QuickLogPreset(label: "Reviewed", entry: "Reviewed output and captured next improvements."),
                QuickLogPreset(label: "Published", entry: "Shared, exported, or archived a finished piece.")]
    case "Learning":
        return [QuickLogPreset(label: "Studied", entry: "Completed a learning session and captured notes."),
                QuickLogPreset(label: "Practiced", entry: "Finished exercises or applied the concept."),
                QuickLogPreset(label: "Reviewed", entry: "Reviewed mistakes, questions, and next lesson.")]
    case "Medicine":
        return [QuickLogPreset(label: "Took dose", entry: "Took today's dose on schedule."),
                QuickLogPreset(label: "Skipped", entry: "Skipped a dose — noted reason."),
                QuickLogPreset(label: "Side effect", entry: "Logged a side effect or interaction."),
                QuickLogPreset(label: "Refilled", entry: "Refilled prescription, captured next refill date.")]
    case "Health":
        return [QuickLogPreset(label: "Vitals", entry: "Logged vitals (BP, weight, sleep, hydration)."),
                QuickLogPreset(label: "Symptom", entry: "Noted a symptom or how I'm feeling today."),
                QuickLogPreset(label: "Doctor", entry: "Doctor visit, results, or follow-up captured.")]
    case "Mindfulness":
        return [QuickLogPreset(label: "Meditated", entry: "Completed a meditation or breathing session."),
                QuickLogPreset(label: "Journaled", entry: "Wrote a reflection or gratitude entry."),
                QuickLogPreset(label: "Mood check", entry: "Captured mood and energy level.")]
    case "Cooking":
        return [QuickLogPreset(label: "Cooked", entry: "Cooked a meal and noted recipe and outcome."),
                QuickLogPreset(label: "Prepped", entry: "Did meal prep and stocked containers."),
                QuickLogPreset(label: "Tasted", entry: "Captured tasting notes and what to tweak.")]
    case "Pets":
        return [QuickLogPreset(label: "Fed", entry: "Fed pet on schedule."),
                QuickLogPreset(label: "Walked", entry: "Took pet for a walk or play session."),
                QuickLogPreset(label: "Vet", entry: "Vet visit, medication, or grooming logged.")]
    case "Travel":
        return [QuickLogPreset(label: "Visited", entry: "Logged a place visited and quick highlights."),
                QuickLogPreset(label: "Spent", entry: "Captured an expense or receipt."),
                QuickLogPreset(label: "Packed", entry: "Updated packing list or trip checklist.")]
    case "Gaming":
        return [QuickLogPreset(label: "Played", entry: "Played a session and noted progress."),
                QuickLogPreset(label: "Achievement", entry: "Unlocked an achievement or milestone."),
                QuickLogPreset(label: "Backlog", entry: "Updated backlog with a new title.")]
    case "Tech":
        return [QuickLogPreset(label: "Shipped", entry: "Shipped a feature, fix, or commit."),
                QuickLogPreset(label: "Debugged", entry: "Tracked down an issue and noted the fix."),
                QuickLogPreset(label: "Learned", entry: "Captured a new tool, library, or pattern.")]
    case "Work":
        return [QuickLogPreset(label: "Deep work", entry: "Completed a deep-work session."),
                QuickLogPreset(label: "Meeting", entry: "Logged outcomes and follow-ups from a meeting."),
                QuickLogPreset(label: "Done", entry: "Closed out a task or deliverable.")]
    case "Finance":
        return [QuickLogPreset(label: "Paid", entry: "Paid a bill or recurring expense."),
                QuickLogPreset(label: "Saved", entry: "Logged a contribution or savings transfer."),
                QuickLogPreset(label: "Reviewed", entry: "Reviewed budget, statements, or subscriptions.")]
    case "Garden":
        return [QuickLogPreset(label: "Watered", entry: "Watered beds and noted soil condition."),
                QuickLogPreset(label: "Harvested", entry: "Harvested produce and logged the haul."),
                QuickLogPreset(label: "Weeded", entry: "Weeded, pruned, or composted today.")]
    default:
        return [QuickLogPreset(label: "Done", entry: "Completed a check-in and logged the result."),
                QuickLogPreset(label: "Checked", entry: "Checked status and captured observations."),
                QuickLogPreset(label: "Updated", entry: "Updated notes and next action.")]
    }
}

File: Android/app/src/main/java/com/swarnkary/donezy/CategoryDefinitions.kt
package com.swarnkary.donezy

import androidx.compose.ui.graphics.Color

// ─── Category definitions ─────────────────────────────────────────────────────

data class CategoryOption(
    val label: String,
    val emoji: String,
    val description: String,
    val starterNotes: String,
    val accent: Color
)

val hobbyCategories = listOf(
    CategoryOption("General",     "📋", "Flexible trackers for anything recurring.",
        "What matters, how often it needs attention, and what a good update looks like.",
        Color(0xFF4E5D6C)),
    CategoryOption("Plants",      "🌱", "Watering, pruning, fertilizer, sunlight, pests.",
        "Plant species, watering preference, light conditions, fertilizer schedule, and warning signs.",
        Color(0xFF2D7A46)),
    CategoryOption("Garden",      "🌻", "Outdoor beds, lawn, compost, harvests, seasonal jobs.",
        "Plot/bed, current crops, watering days, fertilizer or compost cadence, and pest watch.",
        Color(0xFF6E8B3D)),
    CategoryOption("Fitness",     "💪", "Workouts, mobility, recovery, steps, practice streaks.",
        "Routine, target frequency, current baseline, progress marker, and recovery notes.",
        Color(0xFFB14E32)),
    CategoryOption("Mindfulness", "🧘", "Meditation, journaling, breathing, gratitude.",
        "Practice length, technique, intentions, and how you felt before/after.",
        Color(0xFF7E57C2)),
    CategoryOption("Reading",     "📚", "Books, chapters, notes, reading streaks.",
        "Current book, chapter target, themes to capture, and next reading checkpoint.",
        Color(0xFF6F4A8E)),
    CategoryOption("Learning",    "🧠", "Courses, language, coding, certifications.",
        "Topic, resource, practice plan, review checkpoints, and questions to revisit.",
        Color(0xFF4867A5)),
    CategoryOption("Music",       "🎵", "Instrument practice, drills, songs, recordings.",
        "Warmups, technique focus, piece/song, tempo target, and recording notes.",
        Color(0xFF2E6F95)),
    CategoryOption("Creative",    "🎨", "Writing, sketching, photos, side projects.",
        "Project goal, next small action, reference ideas, and review cadence.",
        Color(0xFFB05D6D)),
    CategoryOption("Cooking",     "🍳", "Recipes, meal prep, baking, fermentation, kitchen wins.",
        "Recipe or technique, ingredients, cooking time, taste notes, and what to change next.",
        Color(0xFFE07A1F)),
    CategoryOption("Coffee",      "☕", "Beans, grind, recipes, brew ratios, tasting notes.",
        "Bean, roast date, grinder setting, dose, yield, time, and taste notes.",
        Color(0xFF8A5A2B)),
    CategoryOption("Pets",        "🐾", "Feeding, walks, grooming, vet, training sessions.",
        "Pet name, feeding cadence, walk/play schedule, grooming routine, and vet milestones.",
        Color(0xFF8D6E63)),
    CategoryOption("Aquarium",    "🐠", "Water tests, feeding, cleaning, livestock health.",
        "Tank size, parameters to test, feeding cadence, cleaning routine, and livestock observations.",
        Color(0xFF207A8A)),
    CategoryOption("Travel",      "✈️", "Trips, packing, places visited, expense receipts.",
        "Destination, dates, packing list, must-see places, and cost log.",
        Color(0xFF0097A7)),
    CategoryOption("Gaming",      "🎮", "Game progress, sessions, achievements, backlog.",
        "Title, current chapter/rank, target achievement, session length, and backlog notes.",
        Color(0xFF5C6BC0)),
    CategoryOption("Tech",        "💻", "Devices, side projects, builds, deployments, learning logs.",
        "Project or device, current task, dependencies, blockers, and next milestone.",
        Color(0xFF455A64)),
    CategoryOption("Work",        "💼", "Standups, deliverables, deep-work sessions, follow-ups.",
        "Goal, current task, blockers, time spent, and what's queued for tomorrow.",
        Color(0xFF37474F)),
    CategoryOption("Finance",     "💰", "Bills, budgets, subscriptions, investments, savings goals.",
        "Account/bill name, due date or cadence, target amount, and balance trend.",
        Color(0xFF388E3C)),
    CategoryOption("Collecting",  "🎒", "Sneakers, keyboards, watches, figures, catalog care.",
        "Item details, condition, purchase date, care routine, and rotation notes.",
        Color(0xFF8B5E83)),
    CategoryOption("Maintenance", "🔧", "Home, gear, bike, car, filters, renewals.",
        "Asset, recurring task, parts/tools, last service date, and next service trigger.",
        Color(0xFF6B6D3F))
)

fun categoryFor(label: String): CategoryOption {
    val trimmed = label.trim()
    return hobbyCategories.firstOrNull { it.label.equals(trimmed, ignoreCase = true) }
        ?: hobbyCategories.first().copy(
            label = trimmed.ifBlank { "General" },
            emoji = "📋",
            description = "Custom tracker category.",
            starterNotes = "What to track, when to check in, and what a useful note should include."
        )
}

// ─── Reminder options ────────────────────────────────────────────────────────

data class ReminderOption(val label: String, val hours: Long, val description: String)

val reminderOptions = listOf(
    ReminderOption("None", 0L,   "No reminder"),
    ReminderOption("1h",   1L,   "Quick follow-up"),
    ReminderOption("3h",   3L,   "Later today"),
    ReminderOption("6h",   6L,   "Half day"),
    ReminderOption("12h",  12L,  "Tonight"),
    ReminderOption("1d",   24L,  "Tomorrow"),
    ReminderOption("2d",   48L,  "Two days"),
    ReminderOption("3d",   72L,  "Three days"),
    ReminderOption("1w",   168L, "Weekly"),
    ReminderOption("2w",   336L, "Bi-weekly"),
    ReminderOption("30d",  720L, "Monthly")
)

fun reminderLabel(hours: Long): String =
    reminderOptions.firstOrNull { it.hours == hours }?.description ?: "${hours}h"

// ─── Starter templates ───────────────────────────────────────────────────────

data class HobbyTemplate(
    val title: String,
    val category: String,
    val trackerName: String,
    val notes: String,
    val reminderHours: Long
)

val hobbyTemplates = listOf(
    HobbyTemplate("Plant care",        "Plants",      "Monstera care",
        "Track watering, leaf checks, pruning, fertilizer, and light changes.", 72L),
    HobbyTemplate("Daily meditation",  "Mindfulness", "Morning meditation",
        "Capture practice length, technique, intention, and how you felt before/after.", 24L),
    HobbyTemplate("Coffee dial-in",    "Coffee",      "Espresso recipe",
        "Record bean, dose, yield, grind, shot time, and taste notes.", 0L),
    HobbyTemplate("Cooking journal",   "Cooking",     "Weeknight dinners",
        "Recipe, ingredients, cooking time, taste notes, and what to change next.", 0L),
    HobbyTemplate("Aquarium routine",  "Aquarium",    "Tank maintenance",
        "Log water parameters, feeding, filter checks, cleaning, and livestock health.", 168L),
    HobbyTemplate("Pet care",          "Pets",        "Dog routine",
        "Walks, feeding, training drills, grooming, and vet milestones.", 12L),
    HobbyTemplate("Practice streak",   "Music",       "Guitar practice",
        "Capture warmups, drills, song sections, tempo, and recordings.", 24L),
    HobbyTemplate("Gear rotation",     "Collecting",  "Sneaker rotation",
        "Track wears, cleaning, condition, weather, and storage notes.", 336L),
    HobbyTemplate("Study plan",        "Learning",    "Android learning",
        "Plan lessons, exercises, questions, review dates, and small projects.", 24L),
    HobbyTemplate("Reading list",      "Reading",     "Currently reading",
        "Current book, chapter target, themes to capture, and weekly reading goal.", 24L),
    HobbyTemplate("Side project",      "Tech",        "Side project",
        "Goal, current task, blockers, deploy/test cadence, next milestone.", 48L),
    HobbyTemplate("Workout split",     "Fitness",     "Push/Pull/Legs",
        "Split day, key lifts, weight progression, mobility, recovery score.", 24L),
    HobbyTemplate("Bills tracker",     "Finance",     "Monthly bills",
        "Bill name, due date, autopay, amount, and renewal notes.", 720L),
    HobbyTemplate("Gaming backlog",    "Gaming",      "Backlog progress",
        "Title, current chapter/rank, target achievement, and total play hours.", 168L),
    HobbyTemplate("Trip planner",      "Travel",      "Next trip",
        "Destination, dates, packing list, must-see places, and cost log.", 0L),
    HobbyTemplate("Garden bed",        "Garden",      "Veggie bed",
        "Bed, planted crops, watering days, fertilizer, and harvest notes.", 24L),
    HobbyTemplate("Daily journal",     "Mindfulness", "Daily journal",
        "Three things, mood, gratitude, and one intention for tomorrow.", 24L),
    HobbyTemplate("Work standup",      "Work",        "Daily standup",
        "Done yesterday, doing today, blockers, and follow-ups owed.", 24L)
)

// ─── Quick-log presets ───────────────────────────────────────────────────────

data class QuickLogPreset(val label: String, val entry: String)

fun quickLogPresetsFor(category: String): List<QuickLogPreset> =
    when (categoryFor(category).label) {
        "Plants"      -> listOf(
            QuickLogPreset("Watered",    "Watered and checked soil moisture."),
            QuickLogPreset("Fertilized", "Added fertilizer and noted plant response."),
            QuickLogPreset("Leaf check", "Checked leaves for pests, yellowing, and new growth."))
        "Fitness"     -> listOf(
            QuickLogPreset("Workout",    "Completed workout and logged effort level."),
            QuickLogPreset("Mobility",   "Completed mobility or stretching session."),
            QuickLogPreset("Recovery",   "Logged rest, soreness, sleep, or recovery notes."))
        "Reading"     -> listOf(
            QuickLogPreset("Read",       "Read a session and captured key takeaways."),
            QuickLogPreset("Quote",      "Saved a memorable idea or quote."),
            QuickLogPreset("Review",     "Reviewed notes and updated next reading target."))
        "Music"       -> listOf(
            QuickLogPreset("Practice",   "Completed practice session with focus notes."),
            QuickLogPreset("Tempo",      "Worked on tempo, timing, or difficult section."),
            QuickLogPreset("Recorded",   "Recorded a take and noted improvements."))
        "Coffee"      -> listOf(
            QuickLogPreset("Brewed",     "Logged brew recipe, extraction, and taste."),
            QuickLogPreset("Adjusted",   "Adjusted grind, dose, yield, or brew time."),
            QuickLogPreset("Tasted",     "Captured tasting notes and next change."))
        "Aquarium"    -> listOf(
            QuickLogPreset("Fed",        "Fed livestock and checked behavior."),
            QuickLogPreset("Tested",     "Tested water parameters and logged results."),
            QuickLogPreset("Cleaned",    "Completed cleaning, water change, or filter check."))
        "Collecting"  -> listOf(
            QuickLogPreset("Used",       "Logged use, wear, condition, and storage notes."),
            QuickLogPreset("Cleaned",    "Cleaned item and checked condition."),
            QuickLogPreset("Cataloged",  "Updated catalog details or purchase notes."))
        "Maintenance" -> listOf(
            QuickLogPreset("Serviced",   "Completed maintenance task and noted parts/tools."),
            QuickLogPreset("Inspected",  "Inspected condition and captured follow-up items."),
            QuickLogPreset("Renewed",    "Updated renewal, warranty, or replacement details."))
        "Creative"    -> listOf(
            QuickLogPreset("Drafted",    "Moved the project forward with a focused work session."),
            QuickLogPreset("Reviewed",   "Reviewed output and captured next improvements."),
            QuickLogPreset("Published",  "Shared, exported, or archived a finished piece."))
        "Learning"    -> listOf(
            QuickLogPreset("Studied",    "Completed a learning session and captured notes."),
            QuickLogPreset("Practiced",  "Finished exercises or applied the concept."),
            QuickLogPreset("Reviewed",   "Reviewed mistakes, questions, and next lesson."))
        "Mindfulness" -> listOf(
            QuickLogPreset("Meditated",  "Completed a meditation or breathing session."),
            QuickLogPreset("Journaled",  "Wrote a reflection or gratitude entry."),
            QuickLogPreset("Mood check", "Captured mood and energy level."))
        "Cooking"     -> listOf(
            QuickLogPreset("Cooked",     "Cooked a meal and noted recipe and outcome."),
            QuickLogPreset("Prepped",    "Did meal prep and stocked containers."),
            QuickLogPreset("Tasted",     "Captured tasting notes and what to tweak."))
        "Pets"        -> listOf(
            QuickLogPreset("Fed",        "Fed pet on schedule."),
            QuickLogPreset("Walked",     "Took pet for a walk or play session."),
            QuickLogPreset("Vet",        "Vet visit, grooming, or care note logged."))
        "Travel"      -> listOf(
            QuickLogPreset("Visited",    "Logged a place visited and quick highlights."),
            QuickLogPreset("Spent",      "Captured an expense or receipt."),
            QuickLogPreset("Packed",     "Updated packing list or trip checklist."))
        "Gaming"      -> listOf(
            QuickLogPreset("Played",     "Played a session and noted progress."),
            QuickLogPreset("Achievement","Unlocked an achievement or milestone."),
            QuickLogPreset("Backlog",    "Updated backlog with a new title."))
        "Tech"        -> listOf(
            QuickLogPreset("Shipped",    "Shipped a feature, fix, or commit."),
            QuickLogPreset("Debugged",   "Tracked down an issue and noted the fix."),
            QuickLogPreset("Learned",    "Captured a new tool, library, or pattern."))
        "Work"        -> listOf(
            QuickLogPreset("Deep work",  "Completed a deep-work session."),
            QuickLogPreset("Meeting",    "Logged outcomes and follow-ups from a meeting."),
            QuickLogPreset("Done",       "Closed out a task or deliverable."))
        "Finance"     -> listOf(
            QuickLogPreset("Paid",       "Paid a bill or recurring expense."),
            QuickLogPreset("Saved",      "Logged a contribution or savings transfer."),
            QuickLogPreset("Reviewed",   "Reviewed budget, statements, or subscriptions."))
        "Garden"      -> listOf(
            QuickLogPreset("Watered",    "Watered beds and noted soil condition."),
            QuickLogPreset("Harvested",  "Harvested produce and logged the haul."),
            QuickLogPreset("Weeded",     "Weeded, pruned, or composted today."))
        else -> listOf(
            QuickLogPreset("Done",       "Completed a check-in and logged the result."),
            QuickLogPreset("Checked",    "Checked status and captured observations."),
            QuickLogPreset("Updated",    "Updated notes and next action."))
    }
